CONTRIBUTING.md

```diff
@@ -23,12 +29,129 @@ tox run -e integration   # integration tests
 tox                      # runs 'format', 'lint', 'static', and 'unit' environments
 ```
 
-## Build the charm
+### Committing
 
-Build the charm in this git repository using:
+This repo uses CI/CD workflows as outlined by
+[operator-workflows](https://github.com/canonical/operator-workflows). The four
+workflows are as follows:
 
-```shell
-charmcraft pack
+- `test.yaml`: This is a series of tests including linting, unit tests and
+  library checks which run on every pull request.
+- `integration_test.yaml`: This runs the suite of integration tests included
+  with the charm and runs on every pull request.
+- `test_and_publish_charm.yaml`: This runs either by manual dispatch or on every
+  push to the main branch or a special track/\*\* branch. Once a PR is merged
+  with one of these branches, this workflow runs to ensure the tests have passed
+  before building the charm and publishing the new version to the edge channel
+  on Charmhub.
+- `promote_charm.yaml`: This is a manually triggered workflow which publishes
+  the charm currently on the edge channel to the stable channel on Charmhub.
+
+These tests validate extensive linting and formatting rules. Before creating a
+PR, please run `tox` to ensure proper formatting and linting is performed.
+
+### Deploy
+
+This charm is used to deploy Airbyte server in a k8s cluster. For a local
+deployment, follow the following steps:
+
+#### Install Microk8s
```

Comment:
Another idea for docs, I'm still hazy on what the recommended approach is for charm docs. Some charms like [observability](https://charmhub.io/topics/canonical-observability-stack) and [identity](https://charmhub.io/topics/canonical-identity-platform) team have their docs on charmhub. But the company read-the-docs is really nice as it lets you write docs in a dedicated repo with better syntax and formatting support, better CI/CD dedicated to docs, etc.

Reply:
The docs here will eventually follow a similar path to the [Temporal charm docs](https://charmhub.io/temporal-k8s) and live on Charmhub.